### ISSUE TITLE
GROUP can't subscribe to dead event.

### ISSUE THREAD

**Issue by JohannesStrauTHBingen** [FlightControl-Master/MOOSE#548]
Wenn a group gets killed the event handler is not calling the on event function.

**Comment by FlightControl-Master** [comment]
Ok. Ill check this!

**Comment by FlightControl-Master** [comment]
I re-tested this and I can't resimulate this problem.
See test mission [EVT-205 - GROUP OnEventDead Example](https://github.com/FlightControl-Master/MOOSE_MISSIONS/tree/master/EVT%20-%20Event%20Handling/EVT-205%20-%20GROUP%20OnEventDead%20Example).
2 planes are flying into 2 groups of battle tanks. The first group (Tank A) has a subscription to process DEAD events, the second group (Tank B) does not.

This is the code:
```lua
TanksGroup = GROUP:FindByName( "Group Tanks A" )

TanksGroup:HandleEvent( EVENTS.Dead )

function TanksGroup:OnEventDead( EventData )

  self:E( "I just got dead and I am part of " .. EventData.IniGroupName )
  EventData.IniUnit:MessageToAll( "I just got dead and I am part of " .. EventData.IniGroupName, 15, "Alert!" )
end
```

This is the output from the dcs.log:
```
00907.715 INFO    SCRIPTING:   1040(    13)/E:               EVENT00000.onEvent({[1]="S_EVENT_SHOT",[2]={["id"]=1,["weapon"]={["id_"]=16779776,},["time"]=43221,["initiator"]={["id_"]=16777472,},},})
00908.860 INFO    SCRIPTING:   1040(    13)/E:               EVENT00000.onEvent({[1]="S_EVENT_SHOT",[2]={["id"]=1,["weapon"]={["id_"]=16780032,},["time"]=43225.58,["initiator"]={["id_"]=16777728,},},})
00911.343 INFO    SCRIPTING:   1040(    13)/E:               EVENT00000.onEvent({[1]="S_EVENT_HIT",[2]={["time"]=43234.8,["initiator"]={["id_"]=16777472,},["target"]={["id_"]=16777984,},["id"]=2,["weapon"]={["id_"]=16779776,},},})
00911.346 INFO    SCRIPTING:    883(    13)/E:               EVENT00000.onEvent({[1]="S_EVENT_DEAD",[2]={["IniUnit"]={["UnitName"]="Tank",["ClassName"]="UNIT",},["IniCoalition"]=2,["initiator"]={["id_"]=16777984,},["id"]=8,["IniTypeName"]="M-2 Bradley",["IniObjectCategory"]=1,["IniUnitName"]="Tank",["IniCategory"]=2,["IniGroupName"]="Group Tanks A",["time"]=43234.8,["IniDCSGroup"]={["id_"]=80,},["IniGroup"]={["GroupName"]="Group Tanks A",["ClassName"]="GROUP",},["IniDCSUnitName"]="Tank",["IniDCSUnit"]=,["IniDCSGroupName"]="Group Tanks A",},[3]="Tank",[5]=-1,})
00911.346 INFO    SCRIPTING:    983(    13)/E:               EVENT00000.onEvent({[1]="Calling OnEventDead for GROUP ",[2]="GROUP#000000036",[3]=4,})
00911.346 INFO    SCRIPTING:     24(    -1)/E:               GROUP00036.function(I just got dead and I am part of Group Tanks A)
00911.346 INFO    SCRIPTING:   1008(    13)/E:               EVENT00000.onEvent({[1]="Calling EventFunction for Class ",[2]="DATABASE#000000003",[3]=1,})
00912.386 INFO    SCRIPTING:   1040(    13)/E:               EVENT00000.onEvent({[1]="S_EVENT_SHOT",[2]={["id"]=1,["weapon"]={["id_"]=16780288,},["time"]=43238.8,["initiator"]={["id_"]=16777472,},},})
00912.539 INFO    SCRIPTING:   1040(    13)/E:               EVENT00000.onEvent({[1]="S_EVENT_HIT",[2]={["time"]=43239.34,["initiator"]={["id_"]=16777728,},["target"]={["id_"]=16779264,},["id"]=2,["weapon"]={["id_"]=16780032,},},})
00912.541 INFO    SCRIPTING:    883(    13)/E:               EVENT00000.onEvent({[1]="S_EVENT_DEAD",[2]={["IniUnit"]={["UnitName"]="Unit #010",["ClassName"]="UNIT",},["IniCoalition"]=2,["initiator"]={["id_"]=16779264,},["id"]=8,["IniTypeName"]="M-2 Bradley",["IniObjectCategory"]=1,["IniUnitName"]="Unit #010",["IniCategory"]=2,["IniGroupName"]="Group Tanks B",["time"]=43239.34,["IniDCSGroup"]={["id_"]=83,},["IniGroup"]={["GroupName"]="Group Tanks B",["ClassName"]="GROUP",},["IniDCSUnitName"]="Unit #010",["IniDCSUnit"]=,["IniDCSGroupName"]="Group Tanks B",},[3]="Unit #010",[5]=-1,})
00912.541 INFO    SCRIPTING:   1008(    13)/E:               EVENT00000.onEvent({[1]="Calling EventFunction for Class ",[2]="DATABASE#000000003",[3]=1,})
00913.506 INFO    SCRIPTING:   1040(    13)/E:               EVENT00000.onEvent({[1]="S_EVENT_SHOT",[2]={["id"]=1,["weapon"]={["id_"]=16780544,},["time"]=43243.34,["initiator"]={["id_"]=16777728,},},})
00915.389 INFO    SCRIPTING:   1040(    13)/E:               EVENT00000.onEvent({[1]="S_EVENT_HIT",[2]={["time"]=43250.16,["initiator"]={["id_"]=16777472,},["target"]={["id_"]=16778240,},["id"]=2,["weapon"]={["id_"]=16780288,},},})
00915.390 INFO    SCRIPTING:    883(    13)/E:               EVENT00000.onEvent({[1]="S_EVENT_DEAD",[2]={["IniUnit"]={["UnitName"]="Unit #001",["ClassName"]="UNIT",},["IniCoalition"]=2,["initiator"]={["id_"]=16778240,},["id"]=8,["IniTypeName"]="M-2 Bradley",["IniObjectCategory"]=1,["IniUnitName"]="Unit #001",["IniCategory"]=2,["IniGroupName"]="Group Tanks A",["time"]=43250.16,["IniDCSGroup"]={["id_"]=80,},["IniGroup"]={["GroupName"]="Group Tanks A",["ClassName"]="GROUP",},["IniDCSUnitName"]="Unit #001",["IniDCSUnit"]=,["IniDCSGroupName"]="Group Tanks A",},[3]="Unit #001",[5]=-1,})
00915.391 INFO    SCRIPTING:    983(    13)/E:               EVENT00000.onEvent({[1]="Calling OnEventDead for GROUP ",[2]="GROUP#000000036",[3]=4,})
00915.391 INFO    SCRIPTING:     24(    -1)/E:               GROUP00036.function(I just got dead and I am part of Group Tanks A)
00915.391 INFO    SCRIPTING:   1008(    13)/E:               EVENT00000.onEvent({[1]="Calling EventFunction for Class ",[2]="DATABASE#000000003",[3]=1,})
00915.509 INFO    SCRIPTING:   1040(    13)/E:               EVENT00000.onEvent({[1]="S_EVENT_SHOOTING_START",[2]={["id"]=22,["target"]={["id_"]=16777472,},["time"]=43250.707,["initiator"]={["id_"]=16778752,},},})
00915.634 INFO    SCRIPTING:   1040(    13)/E:               EVENT00000.onEvent({[1]="S_EVENT_SHOOTING_START",[2]={["id"]=22,["target"]={["id_"]=16777472,},["time"]=43251.217,["initiator"]={["id_"]=16779008,},},})
00915.725 INFO    SCRIPTING:   1040(    13)/E:               EVENT00000.onEvent({[1]="S_EVENT_SHOOTING_END",[2]={["id"]=23,["time"]=43251.606,["initiator"]={["id_"]=16778752,},},})
00915.853 INFO    SCRIPTING:   1040(    13)/E:               EVENT00000.onEvent({[1]="S_EVENT_SHOOTING_END",[2]={["id"]=23,["time"]=43252.123,["initiator"]={["id_"]=16779008,},},})
00916.011 INFO    SCRIPTING:   1040(    13)/E:               EVENT00000.onEvent({[1]="S_EVENT_SHOOTING_START",[2]={["id"]=22,["target"]={["id_"]=16777728,},["time"]=43252.741,["initiator"]={["id_"]=16778496,},},})
00916.180 INFO    SCRIPTING:   1040(    13)/E:               EVENT00000.onEvent({[1]="S_EVENT_SHOOTING_END",[2]={["id"]=23,["time"]=43253.341,["initiator"]={["id_"]=16778496,},},})
00916.380 INFO    SCRIPTING:   1040(    13)/E:               EVENT00000.onEvent({[1]="S_EVENT_SHOT",[2]={["id"]=1,["weapon"]={["id_"]=16780800,},["time"]=43254.16,["initiator"]={["id_"]=16777472,},},})
00916.569 INFO    SCRIPTING:   1040(    13)/E:               EVENT00000.onEvent({[1]="S_EVENT_HIT",[2]={["time"]=43254.82,["initiator"]={["id_"]=16777728,},["target"]={["id_"]=16779008,},["id"]=2,["weapon"]={["id_"]=16780544,},},})
00916.571 INFO    SCRIPTING:    883(    13)/E:               EVENT00000.onEvent({[1]="S_EVENT_DEAD",[2]={["IniUnit"]={["UnitName"]="Unit #009",["ClassName"]="UNIT",},["IniCoalition"]=2,["initiator"]={["id_"]=16779008,},["id"]=8,["IniTypeName"]="M-2 Bradley",["IniObjectCategory"]=1,["IniUnitName"]="Unit #009",["IniCategory"]=2,["IniGroupName"]="Group Tanks B",["time"]=43254.82,["IniDCSGroup"]={["id_"]=83,},["IniGroup"]={["GroupName"]="Group Tanks B",["ClassName"]="GROUP",},["IniDCSUnitName"]="Unit #009",["IniDCSUnit"]=,["IniDCSGroupName"]="Group Tanks B",},[3]="Unit #009",[5]=-1,})
00916.571 INFO    SCRIPTING:   1008(    13)/E:               EVENT00000.onEvent({[1]="Calling EventFunction for Class ",[2]="DATABASE#000000003",[3]=1,})
00916.856 INFO    SCRIPTING:   1040(    13)/E:               EVENT00000.onEvent({[1]="S_EVENT_SHOOTING_START",[2]={["id"]=22,["target"]={["id_"]=16777472,},["time"]=43255.946,["initiator"]={["id_"]=16778752,},},})
00917.071 INFO    SCRIPTING:   1040(    13)/E:               EVENT00000.onEvent({[1]="S_EVENT_SHOOTING_END",[2]={["id"]=23,["time"]=43256.838,["initiator"]={["id_"]=16778752,},},})
00917.369 INFO    SCRIPTING:   1040(    13)/E:               EVENT00000.onEvent({[1]="S_EVENT_SHOOTING_START",[2]={["id"]=22,["target"]={["id_"]=16777728,},["time"]=43257.932,["initiator"]={["id_"]=16778496,},},})
00917.590 INFO    SCRIPTING:   1040(    13)/E:               EVENT00000.onEvent({[1]="S_EVENT_SHOT",[2]={["id"]=1,["weapon"]={["id_"]=16781056,},["time"]=43258.82,["initiator"]={["id_"]=16777728,},},})
00917.595 INFO    SCRIPTING:   1040(    13)/E:               EVENT00000.onEvent({[1]="S_EVENT_SHOOTING_END",[2]={["id"]=23,["time"]=43258.836,["initiator"]={["id_"]=16778496,},},})
00918.232 INFO    SCRIPTING:   1040(    13)/E:               EVENT00000.onEvent({[1]="S_EVENT_SHOT",[2]={["id"]=1,["weapon"]={["id_"]=16781312,},["time"]=43261.14,["initiator"]={["id_"]=16777472,},},})
00918.335 INFO    SCRIPTING:   1040(    13)/E:               EVENT00000.onEvent({[1]="S_EVENT_SHOT",[2]={["id"]=1,["weapon"]={["id_"]=16781568,},["time"]=43261.44,["initiator"]={["id_"]=16777472,},},})
00918.615 INFO    SCRIPTING:   1040(    13)/E:               EVENT00000.onEvent({[1]="S_EVENT_HIT",[2]={["time"]=43262.24,["initiator"]={["id_"]=16777472,},["target"]={["id_"]=16778496,},["id"]=2,["weapon"]={["id_"]=16780800,},},})
00918.617 INFO    SCRIPTING:    883(    13)/E:               EVENT00000.onEvent({[1]="S_EVENT_DEAD",[2]={["IniUnit"]={["UnitName"]="Unit #002",["ClassName"]="UNIT",},["IniCoalition"]=2,["initiator"]={["id_"]=16778496,},["id"]=8,["IniTypeName"]="M-2 Bradley",["IniObjectCategory"]=1,["IniUnitName"]="Unit #002",["IniCategory"]=2,["IniGroupName"]="Group Tanks A",["time"]=43262.24,["IniDCSGroup"]={["id_"]=80,},["IniGroup"]={["GroupName"]="Group Tanks A",["ClassName"]="GROUP",},["IniDCSUnitName"]="Unit #002",["IniDCSUnit"]=,["IniDCSGroupName"]="Group Tanks A",},[3]="Unit #002",[5]=-1,})
00918.617 INFO    SCRIPTING:    983(    13)/E:               EVENT00000.onEvent({[1]="Calling OnEventDead for GROUP ",[2]="GROUP#000000036",[3]=4,})
00918.617 INFO    SCRIPTING:     24(    -1)/E:               GROUP00036.function(I just got dead and I am part of Group Tanks A)
00918.617 INFO    SCRIPTING:   1008(    13)/E:               EVENT00000.onEvent({[1]="Calling EventFunction for Class ",[2]="DATABASE#000000003",[3]=1,})
00919.731 INFO    SCRIPTING:   1040(    13)/E:               EVENT00000.onEvent({[1]="S_EVENT_SHOT",[2]={["id"]=1,["weapon"]={["id_"]=16781824,},["time"]=43265.72,["initiator"]={["id_"]=16777728,},},})
00919.833 INFO    SCRIPTING:   1040(    13)/E:               EVENT00000.onEvent({[1]="S_EVENT_SHOT",[2]={["id"]=1,["weapon"]={["id_"]=16782080,},["time"]=43266.02,["initiator"]={["id_"]=16777728,},},})
00920.110 INFO    SCRIPTING:   1040(    13)/E:               EVENT00000.onEvent({[1]="S_EVENT_HIT",[2]={["time"]=43266.86,["initiator"]={["id_"]=16777728,},["target"]={["id_"]=16778752,},["id"]=2,["weapon"]={["id_"]=16781056,},},})
00920.113 INFO    SCRIPTING:    883(    13)/E:               EVENT00000.onEvent({[1]="S_EVENT_DEAD",[2]={["IniUnit"]={["UnitName"]="Tank #001",["ClassName"]="UNIT",},["IniCoalition"]=2,["initiator"]={["id_"]=16778752,},["id"]=8,["IniTypeName"]="M-2 Bradley",["IniObjectCategory"]=1,["IniUnitName"]="Tank #001",["IniCategory"]=2,["IniGroupName"]="Group Tanks B",["time"]=43266.86,["IniDCSGroup"]={["id_"]=83,},["IniGroup"]={["GroupName"]="Group Tanks B",["ClassName"]="GROUP",},["IniDCSUnitName"]="Tank #001",["IniDCSUnit"]=,["IniDCSGroupName"]="Group Tanks B",},[3]="Tank #001",[5]=-1,})
00920.113 INFO    SCRIPTING:   1008(    13)/E:               EVENT00000.onEvent({[1]="Calling EventFunction for Class ",[2]="DATABASE#000000003",[3]=1,})
00928.249 INFO    SCRIPTING:   1040(    13)/E:               EVENT00000.onEvent({[1]="S_EVENT_MISSION_END",[2]={["id"]=12,["time"]=43294.034,},})
00928.260 INFO    Config: netview stopped
```

**Comment by FlightControl-Master** [comment]
Check above @JohannesStrauTHBingen . Closing this issue until further notice.

**Comment by Delta-99** [comment]
Was this for plane groups? If so, might want to try that. I think not all plane groups throw the DEAD event. Like if they Eject.

**Comment by FlightControl-Master** [comment]
When they eject, there is a separate event for that.

**Comment by Delta-99** [comment]
Yes, but I believe I found that if a pilot EJECTS there will be no DEAD event. So just pointing that out in case it makes a difference for some situations. I believe there is always a CRASH event though cause the plane either breaks up and crashes or crashes to the ground and breaks up. I could be wrong on this so testing will be needed but I'm pretty sure while I was debugging like crazy this is what I saw.
